### PR TITLE
Fix typo in web3

### PR DESCRIPTION
**Pull request by asgeir-s**
Just updated "web3.ssh" to "web3.shh".

### CODE REVIEW

**Comment by Levino** [comment]
Duplicate of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/28606

What I said there holds for this PR as well. Please add a regression test.

**Review by levino — CHANGES_REQUESTED** [review]
Please add a regression test.

**Review by phra — APPROVED** [review]

**Comment by Levino** [comment]
I will review tomorrow. Looks good on mobile. Thank you for the contribution.

File: types/web3/types.d.ts [hunk 1]
```diff
@@ -85,5 +85,10 @@ export interface Subscribe<T> {
     on(type: "error", handler: (data: Error) => void): void;
 }
 
-export class Shh {} // TODO: Type
+export interface Shh {
+    generateSymKeyFromPassword(
+        password: string,
+        callback?: Callback<string>
+    ): Promise<string>;
```
Comment (Levino) [inline types/web3/types.d.ts]: I guess the [docs](https://web3js.readthedocs.io/en/1.0/web3-shh.html?highlight=generateSymKeyFromPassword#id61) are incorrect here and this functions returns a promise for a string, not a string. Question: Does it also return a promise when a callback is provided? I would say an overload would be more appropriate here. I created an [issue](https://github.com/ethereum/web3.js/issues/1924). Even if a promise is returned when a `callback` parameter is given, we should not type this bad behaviour here. So I would say we make two versions of the this function. One which takes just a string and returns a `Promise<string>` and one which takes a string and a callback function and returns `void`.

File: types/web3/web3-tests.ts [hunk 2]
```diff
@@ -6,12 +6,12 @@ web3.setProvider(myProvider);
...
-	{
-		from: "0x1234567890123456789012345678901234567891",
-		gasPrice: "20000000000"
-	}
+    [],
+    "0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe",
+    {
+        from: "0x1234567890123456789012345678901234567891",
+        gasPrice: "20000000000"
+    }
```
Comment (Levino) [inline types/web3/web3-tests.ts]: Please undo the whitespace changes.

File: types/web3/web3-tests.ts [hunk 3]
```diff
@@ -21,3 +21,6 @@ myContract.options.gas = 5000000;
 const weiStr: string = web3.utils.toWei("100", "gwei");
 const weiBn: BigNumber = web3.utils.toWei(web3.utils.toBN("1"));
 const rndHex: string = Web3.utils.randomHex(10);
+web3.shh
+    .generateSymKeyFromPassword("Never use this password - password!")
+    .then(console.log);
```
Comment (Levino) [inline types/web3/web3-tests.ts]: Instead I would prefer
```ts
const sshPromise: Promise<string> = web3.ssh.generateSymKeyFromPassword("xyz")
```
Keep it simple.

**Review by levino — CHANGES_REQUESTED** [review]
Please address my remarks.

File: types/web3/types.d.ts [hunk 4]
```diff
@@ -85,5 +85,10 @@ export interface Subscribe<T> {
...
+} // TODO: Type
```
[the remark this answers is not in the document]
Comment (Levino) [reply types/web3/types.d.ts]: Maybe `// TODO: type every method` would be more correct.

[the remark this answers is not in the document]
Comment (zlumer) [reply types/web3/types.d.ts]: Interface typings are far from done, so +1 on wording change.

File: types/web3/types.d.ts (hunk 1 above)
Comment (zlumer) [reply types/web3/types.d.ts]: Fully agreed.

**Review by levino — CHANGES_REQUESTED** [review]
Nice. Thanks!

**Review by levino — APPROVED** [review]
Misclick.

File: types/web3/web3-tests.ts [hunk 5]
```diff
@@ -21,3 +21,4 @@ myContract.options.gas = 5000000;
 const weiStr: string = web3.utils.toWei("100", "gwei");
 const weiBn: BigNumber = web3.utils.toWei(web3.utils.toBN("1"));
 const rndHex: string = Web3.utils.randomHex(10);
+const sshPromise: Promise<string> = web3.shh.generateSymKeyFromPassword("xyz");
```
Comment (yxliang01) [inline types/web3/web3-tests.ts]: Is this `shh` -> `ssh` typo again?

**Review by yxliang01 — CHANGES_REQUESTED** [review]

**Review by levino — APPROVED** [review]

File: types/web3/web3-tests.ts (hunk 5 above)
Comment (yxliang01) [reply types/web3/web3-tests.ts]: @Levino haha Lol

**Review by yxliang01 — APPROVED** [review]
LGTM

**Pull request merged**